cinch/roles/jenkins_master/tasks/install.yml

```diff
@@ -1,3 +1,10 @@
+- name: unpin jenkins package for version upgrades
+  lineinfile:
+    dest: "{{ version_pin_file }}"
+    state: absent
+    regexp: "^exclude=jenkins$"
```

Comment:
This will only find lines that are exclusively excluding Jenkins. I'm not certain about the pin file format - is there only one package per exclude line, or can a line read ``exclude=jenkins kernel glibc``?

Reply:
Yeah, you can provide a space-separated list.

"exclude List of packages to exclude from updates or installs. This should be a space separated list. Shell globs using wildcards (eg. * and ?) are allowed."

https://linux.die.net/man/5/yum.conf

Reply:
Let's try something more like this, then:
``
lineinfile:
  dest: "{{ version_pin_file }}"
  state: present
  regexp: "^exclude=(.*)jenkins(.*)$"
  backrefs: true
  line: "exclude=\1 \2"
``

According to the Ansible documentation, when you have ``backrefs`` set to true and ``state: present``, then things will only be changed if ``regexp`` matches an existing line. If the user has no exclude line that contains Jenkins, then the file will not be modified. However, if they do, this /should/ preserve the value of their exclusions while still allowing us to upgrade Jenkins.

As an alternative, if we want to force them to use the value of ``upgrade_blacklist``, we can go with something like
``
lineinfile:
  dest: "{{ version_pin_file }}"
  state: present
  regexp: "^exclude=.*"
  line: "exclude={{ upgrade_blacklist | difference(['jenkins']) | join(' ') }}"
``

This would replace whatever the exclude line is with the values currently configured in upgrade_blacklist (regardless of any manual edits) but omit the "jenkins" entry.

Reply:
Please note that I haven't tested the above plays, I'm just shooting from the hip, here.

Reply:
Thanks for the suggestions.  I had some trouble with finding a regex I was happy with to avoid extraneous spaces, and the second suggestion works well based on my testing.  I'm fine with the dependency on the upgrade_blacklist, and I've updated the comment on the jenkins_upgrade variable to account for this.